client/app/scripts/superdesk-highlights/highlights.js

```diff
@@ -250,27 +250,67 @@
+    function PackageHighlightsDropdownDirective(superdesk, desks, highlightsService, $location, $route) {
         return {
             templateUrl: 'scripts/superdesk-highlights/views/package_highlights_dropdown_directive.html',
             link: function(scope) {
-
-                scope.createHighlight = function(highlight) {
-                    highlightsService.createEmptyHighlight(highlight)
-                    .then(function(new_package) {
-                        superdesk.intent('author', 'package', new_package);
+                scope.$watch(function() {
+                    return desks.active;
+                }, function(active) {
+                    scope.selected = active;
+                    highlightsService.get(desks.getCurrentDeskId()).then(function(result) {
+                        scope.highlights = result._items;
+                        scope.hasHighlights = _.size(scope.highlights) > 0;
                     });
+                });
+
+                scope.listHighlight = function(highlight) {
+                    $location.url('workspace/highlights?highlight=' + highlight._id);
+                    $route.reload();
                 };
+            }
+        };
+    }
 
+    HighlightLabelDirective.$inject = ['desks', 'highlightsService'];
+    function HighlightLabelDirective(desks, highlightsService) {
+        return {
+            scope: {highlight_id: '=highlight'},
+            template: '<span translate>{{ highlightItem.name }}</span>',
+            replate: true,
+            link: function(scope) {
                 highlightsService.get(desks.getCurrentDeskId()).then(function(result) {
-                    scope.highlights = result._items;
-                    scope.hasHighlights = _.size(scope.highlights) > 0;
+                    scope.highlightItem =  _.find(result._items, {_id: scope.highlight_id});
                 });
             }
         };
     }
 
+    CreateHighlightsButtonDirective.$inject = ['superdesk', 'desks', 'highlightsService', '$location'];
+    function CreateHighlightsButtonDirective(superdesk, desks, highlightsService, $location) {
+        return {
+            require: ['^sdAuthoringContainer'],
+            scope: {highlight_id: '=highlight'},
+            templateUrl: 'scripts/superdesk-highlights/views/create_highlights_button_directive.html',
+            link: function(scope, elem, attrs, ctrls) {
+                var authoring = ctrls[0];
+
+                scope.createHighlight = function(highlight) {
+                    highlightsService.get(desks.getCurrentDeskId()).then(function(result) {
+                    scope.highlights = _.find(result._items, {_id: scope.highlight_id});
+                    scope.hasHighlights = _.size(scope.highlight) > 0;
+                }).then(function() {
+                    highlightsService.createEmptyHighlight(scope.highlights)
+                    .then(function(new_package) {
```

Comment:
It's a bit more readable if the promise chain is flat, rather than having nested `.then()` calls, IMO.